Mathlib/CategoryTheory/Galois/Decomposition.lean

```diff
@@ -0,0 +1,206 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.CategoryTheory.Galois.Basic
+
+/-!
+# Decomposition of objects into connected components and applications
+
+We show that in a Galois category every object is the (finite) coproduct of connected subobjects.
+This has many useful corollaries, in particular that the fiber of every object
+is represented by a Galois object (TODO).
+
+## Main results
+
+* `has_decomp_connected_components`: Every object is the sum of its (finitely many) connected
+  components.
+* `fiber_in_connected_component`: An element of the fiber of `X` lies in the fiber of some
+  connected component.
+* `connected_component_unique`: Up to isomorphism, for each element `x` in the fiber of `X` there
+  is only one connected component whose fiber contains `x`.
+
+-/
+
+universe u₁ u₂ w
+
+namespace CategoryTheory
+
+open Limits Functor
+
+variable {C : Type u₁} [Category.{u₂} C]
+
+namespace Cofan
+
+variable {ι₁ ι₂ : Type} {X X₁ X₂ : C} {f₁ : ι₁ → C} {f₂ : ι₂ → C}
+    (g₁ : (i : ι₁) → (f₁ i) ⟶ X₁) (g₂ : (i : ι₂) → (f₂ i) ⟶ X₂) (v₁ : X₁ ⟶ X) (v₂ : X₂ ⟶ X)
+    (h₁ : IsColimit (Cofan.mk X₁ g₁)) (h₂ : IsColimit (Cofan.mk X₂ g₂))
```

Comment:
Instead of using variables `X₁`, `g₁`, etc, it would be better to have a variable `c₁ : Cofan f₁` and `h₁ : IsColimit c₁`. A general `BinaryCofan` instead of a `BinaryCofan.mk` would also be better.